### ISSUE TITLE
Potential improvement for the Application Menu Spam in Gnome

### ISSUE THREAD

**Issue by sdwolfz** [sadko4u/lsp-plugins#285]
On Gnome, you have this situation when browsing the application list:

![image](https://user-images.githubusercontent.com/3746963/208516264-d777f14c-4477-4220-96c7-0b3a0a250d22.png)

The `.desktop` entries for LSP are not grouped together like they are for the main menu:

![image](https://user-images.githubusercontent.com/3746963/208516464-7a25f019-542d-43e8-b1b6-2de106d24dce.png)

After digging around a bit I've discovered that Gnome indeed respects XDG settings, but for some reason it does not use `/etc/xdg/menus/applications-merged/lsp-plugins.menu`

For example here is how `/usr/share/desktop-directories/Utility.directory` is interpreted:
![image](https://user-images.githubusercontent.com/3746963/208516694-0d589d85-f341-4356-8195-fa8f30cf22fe.png)

LSP Plugins menu contains this:

```xml
<Menu>
  <Name>Applications</Name>
  <Menu>
    <Name>Multimedia</Name>
    <Menu>
      <Name>lsp-plugins</Name>
      <Directory>lsp-plugins.directory</Directory>
      <Include>
        <Filename>in.lsp_plug.lsp_plugins_art_delay_mono.desktop</Filename>
...
```

It's referencing items by filename, which makes it more difficult to manually work with.

To make it easier for end users to define their own grouping in dconf, my proposal would be to add a category in the `.desktop` entries for all plugins:

```
[Desktop Entry]
Version=1.0
Type=Application
Name=Mono Limiter
GenericName=Limiter
GenericName[ru]=Лимитер
Comment=Implements a brick-wall limiter with flexible configuration. It prevents input monosignal from raising over the specified Threshold.
Comment[ru]=Реализует лимитер с жёсткой верхней границей. Предотвращает превышение моносигналом заданного порогового уровня. 
Exec=lsp-plugins-limiter-mono
Icon=lsp-plugins
Terminal=false
StartupNotify=false
Keywords=audio;sound;jackd;lsp-plugins;

# This is the addition:
Category=LinuxStudio
```

This way, end users can quickly fix things by creating this entry in dconf:

![image](https://user-images.githubusercontent.com/3746963/208517871-ed2afa4f-9a65-4c49-a0b2-42591047b7a9.png)

We could also change the `menu` to group by category:

```
<Menu>
  <Name>Applications</Name>
  <Menu>
    <Name>Multimedia</Name>
    <Menu>
      <Name>lsp-plugins</Name>
      <Directory>lsp-plugins.directory</Directory>
      <Include>
          <And>
                  <Category>LinuxStudio</Category>
          </And>
      </Include>
    </Menu>
...
```

While this is not a definite solution, I think it would improve things a bit. I'm willing to open up PRs in all plugin repositories to add these changes myself, but I'd like to hear your opinion first on my proposal.

**Comment by sadko4u** [comment]
Hello!
Looks like a good proposal. I think we can dig more deep into it while working with 1.2.5 release.
FYI: 1.2.4 release is almost done and will be issued within few days.

**Comment by sdwolfz** [comment]
No rush, 1.2.5 is fine. I'll start creating PRs this evening for the category. Just to confirm, are you OK with the name?
```
<Category>LinuxStudio</Category>
```
As for why `lsp-plugins.menu` is ignored, not really sure how to debug this further. The docs here:
* https://help.gnome.org/admin//system-admin-guide/2.32/menustructure-13.html.en
... mention it uses the first `applications.menu` file found, but not sure how extra files are handled, or if they are handled at all.

**Comment by sadko4u** [comment]
I think this is not necessary to create PRs. This will take a lot of time for creating them and, after that, applying them since there is a huge amount of repositories to do the same things.

It would be much better if you could do one PR to one plugin or describe what to do in this topic, and I will quickly commit the changes to all other repositories. Another way, you can take the `devel` branch of LSP Plugins, do the job and then submit the single patch here.

The support of XDG feature in GNOME environment was always frustrating for me.

**Comment by sadko4u** [comment]
Hi! I'm back.

What I see at this moment, the category should start it's name with the 'X-' prefix. For example, 'X-LSP-Plugins' because of this requirement:

> The Categories field is a list of strings used to classify menu items. For example, applications in the AudioVideo category might end up in the "Sound & Video" submenu. Appendix A, Registered Categories enumerates the standard categories. Categories not in this document must be prefixed by the string "X-" indicating that they are extensions. Categories are case-sensitive. 

https://specifications.freedesktop.org/menu-spec/latest/ar01s03.html

**Comment by sadko4u** [comment]
@sdwolfz 

Seems that didn't help. I see the set of applications from the `LSP Plugins` menu in the GNOME:

![изображение](https://user-images.githubusercontent.com/12834464/213525436-88530e1f-ad56-4f2d-8e96-85f3761b63d2.png)

**Comment by sadko4u** [comment]
Adding `NotShowIn=GNOME;` to each desktop entry helps to hide these applications in the GNOME menu.

**Comment by sadko4u** [comment]
Committed changes to all repositories.

**Comment by sdwolfz** [comment]
Sorry I got distracted with real life and forgot about this. Thanks for following up and fixing.

**Comment by sadko4u** [comment]
Available in 1.2.5
